### ISSUE TITLE
ASN.1 compiler - encode crash with jer + maps options

### ISSUE THREAD

**Issue by visciang** [erlang/otp#5129]
**Describe the bug**
The ASN1 compiler produce incorrect enc/dec code when `jer` and `maps` options are used.

**To Reproduce**
Example.erl:
```asn1
Example DEFINITIONS ::= BEGIN
    ASequence ::= SEQUENCE {
        fieldA INTEGER,
        fieldB UTF8String
    }
END
```
Compile and test with:
```
❯ erlc +maps -bjer Example.asn1

❯ erl
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Eshell V12.0.3  (abort with ^G)
1> 'Example':encode('ASequence', #{fieldA => 1, fieldB => "aaa"}).
{error,{asn1,{{{encode,{sequence,'ASequence',2,
                                 [{<<"fieldA">>,'INTEGER',mandatory},
                                  {<<"fieldB">>,string,mandatory}]}},
               #{fieldA => 1,fieldB => "aaa"}},
              [{'Example',encode_jer,2,
                          [{file,"/home/visciang/Code/Example.erl"},
                           {line,372}]},
               {'Example',encode,2,
                          [{file,"/home/visciang/Code/Example.erl"},
                           {line,44}]},
               {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,689}]},
               {shell,exprs,7,[{file,"shell.erl"},{line,686}]},
               {shell,eval_exprs,7,[{file,"shell.erl"},{line,642}]},
               {shell,eval_loop,3,[{file,"shell.erl"},{line,627}]}]}}}
```

The same test pass with the `ber` encoding:

```
❯ erlc +maps -bber Example.asn1

❯ erl                                                                  21:12:32
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]

Eshell V12.0.3  (abort with ^G)
1> 'Example':encode('ASequence', #{fieldA => 1, fieldB => "aaa"}).
{ok,<<48,8,2,1,1,12,3,97,97,97>>}
```

**Expected behavior**
Well, it encodes and decodes.

**Affected versions**
Erlang/OTP 24 [erts-12.0.3]

**Comment by KennethL** [comment]
The `maps` option in combination with the `jer` encoding rules is not implemented at all. Today the `maps` option is just silently ignored for `jer` and the same code as if it was not present is generated. 
I think it would be quite easy to implement support for `maps` in combination with `jer`, I will look into if that is even easier than to add an error handling to the compiler saying that the combination is not supported.

**Comment by KennethL** [comment]
Corrected in #5757 , will be part of OTP 24.3
